### PR TITLE
Removed effect/dummy/phased_mob copypasta. Fixed squeak component triggering from abstract effects & co.

### LINKED ISSUE #54777: directional lights will squeak things that normally squeak when stepped on, like rubber ducks
Recreation steps:

1) Get a rubber ducky. Quack.

2) Shine a torch on it.

3) Look away and then towards the duck. It'll quack.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Title. Two stones, one bird.

## Why It's Good For The Game
This will close #54777, and actually code in a generic phased_mob effect to put an end to the copypasted behaviour of its subtypes.

## Changelog
:cl:
fix: squeaking objects should no longer squeak when crossed by abstract or floating objects such as directional lights.
fix: All forms of phasings shouldn't be able to pass through certain restricted turfs now, not just wiz jaunts.
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
Quack.

**Comment by gitbirb** [comment]
stealth demon buff, rubber ducky nerf. truly saddening.

File: code/modules/mob/living/bloodcrawl.dm [hunk 1]
```diff
@@ -1,30 +1,3 @@
-/obj/effect/dummy/phased_mob/slaughter //Can't use the wizard one, blocked by jaunt/slow
-	name = "water"
-	icon = 'icons/effects/effects.dmi'
-	icon_state = "nothing"
```
Comment (LemonInTheDark) [inline code/modules/mob/living/bloodcrawl.dm:4]: are you dropping this?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
Can you elaborate further on what you're changing with effect/dummy code? Just saying "cleaned it up" worries me, because while some of these subtypes are stupid, some seem intentionally distinct.

File: code/modules/mob/living/bloodcrawl.dm (hunk 1 above)
Comment (Ghommie) [reply code/modules/mob/living/bloodcrawl.dm:4]: There is no "nothing" icon_state in `effects.dmi`, and the same icon file is used in`obj/effect` anyway.

**Comment by Ghommie** [comment]
> 
> Can you elaborate further on what you're changing with effect/dummy code?

While the `effect/dummy/phased_mob` path already exists, it contains no extra code and is only there for typechecks, the common behaviour of all phased_mob effects is actually written down on every subtype.
That said, I'm refactoring that heck so it's not a heap of copypasted code.

File: code/game/objects/effects/phased_mob.dm [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	var/turf/newloc = get_step(src,direction)
+	if(!newloc)
+		return FALSE
+	movedelay = world.time + movespeed
+	if(newloc.flags_1 & NOJAUNT_1)
+		to_chat(user, "<span class='warning'>Some strange aura is blocking the way.</span>")
+		return FALSE
+	return newloc
+
+//used for bloodcrawl.
```
Comment (Rohesie) [inline code/game/objects/effects/phased_mob.dm:44]:
```suggestion
/// React to signals by deleting the effect. Used for bloodcrawl,
```

File: code/game/objects/effects/phased_mob.dm [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	return BULLET_ACT_FORCE_PIERCE
+
+/obj/effect/dummy/phased_mob/relaymove(mob/living/user, direction)
+	var/turf/newloc = phased_check(user, direction)
+	if(!newloc)
+		return
+	setDir(direction)
+	forceMove(newloc)
+
+/obj/effect/dummy/phased_mob/proc/phased_check(mob/living/user, direction)
```
Comment (Rohesie) [inline code/game/objects/effects/phased_mob.dm:31]:
```suggestion

/// Checks if the conditions are valid to be able to phase.
/obj/effect/dummy/phased_mob/proc/phased_check(mob/living/user, direction)
```

File: code/game/objects/effects/phased_mob.dm [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	RETURN_TYPE(/turf)
+	if (movedelay > world.time || !direction)
+		return FALSE
+	var/turf/newloc = get_step(src,direction)
+	if(!newloc)
+		return FALSE
+	movedelay = world.time + movespeed
+	if(newloc.flags_1 & NOJAUNT_1)
+		to_chat(user, "<span class='warning'>Some strange aura is blocking the way.</span>")
+		return FALSE
```
Comment (Rohesie) [inline code/game/objects/effects/phased_mob.dm:41]:
```suggestion
		return
	var/turf/newloc = get_step(src,direction)
	if(!newloc)
		return
	movedelay = world.time + movespeed
	if(newloc.flags_1 & NOJAUNT_1)
		to_chat(user, "<span class='warning'>Some strange aura is blocking the way.</span>")
		return
```
If the truthy return type is a `/turf`, then the falsey needs to be `null`, not the boolean false.

File: code/modules/antagonists/slaughter/slaughterevent.dm [hunk 5]
```diff
@@ -31,7 +31,7 @@
 		message_admins("No valid spawn locations found, aborting...")
 		return MAP_ERROR
 
-	var/obj/effect/dummy/phased_mob/slaughter/holder = new /obj/effect/dummy/phased_mob/slaughter((pick(spawn_locs)))
+	var/obj/effect/dummy/phased_mob/holder = new /obj/effect/dummy/phased_mob((pick(spawn_locs)))
```
Comment (Rohesie) [inline code/modules/antagonists/slaughter/slaughterevent.dm:34]:
```suggestion
	var/obj/effect/dummy/phased_mob/holder = new (pick(spawn_locs))
```

File: code/modules/mob/living/bloodcrawl.dm [hunk 6]
```diff
@@ -54,7 +27,7 @@
 	playsound(get_turf(src), 'sound/magic/enter_blood.ogg', 50, TRUE, -1)
 	// Extinguish, unbuckle, stop being pulled, set our location into the
 	// dummy object
-	var/obj/effect/dummy/phased_mob/slaughter/holder = new /obj/effect/dummy/phased_mob/slaughter(mobloc)
+	var/obj/effect/dummy/phased_mob/holder = new /obj/effect/dummy/phased_mob(mobloc)
```
Comment (Rohesie) [inline code/modules/mob/living/bloodcrawl.dm:30]:
```suggestion
	var/obj/effect/dummy/phased_mob/holder = new (mobloc)
```

File: code/modules/mob/living/bloodcrawl.dm [hunk 7]
```diff
@@ -1,30 +1,3 @@
-/obj/effect/dummy/phased_mob/slaughter //Can't use the wizard one, blocked by jaunt/slow
```
Comment (AnturK) [inline code/modules/mob/living/bloodcrawl.dm:1]: There was a comment about this.

Comment (Ghommie) [reply code/modules/mob/living/bloodcrawl.dm:1]: There is be no reason to use the wizard jaunt path anyway unless you want it to work like it.

Comment (AnturK) [reply code/modules/mob/living/bloodcrawl.dm:1]: What i'm saying you put the nojaunt turf check on base type. Either mention this in the changelog at least or keep the old functionality.

Comment (Ghommie) [reply code/modules/mob/living/bloodcrawl.dm:1]: I'll mention it in the changelog. That check exists to stop mobs from phasing where they shouldn't such as hilbert hotel, multicam and transit space.

**Pull request merged**